src/main/java/org/tugraz/sysds/runtime/util/UtilFunctions.java

```diff
@@ -621,7 +614,14 @@ public static int computeNnz(BitSet a, int ai, int len) {
 			lnnz += a.get(i) ? 1 : 0;
 		return lnnz;
 	}
-	
+
+	public static int computeNnz(String[] a, int ai, int len) {
+		int lnnz = 0;
+		for( int k=ai; k<ai+len; k++ )
+			lnnz += (a[k] != null && !a[k].isEmpty()) ? 1 : 0;
```

Comment:
please also treat all Strings of value zero (0, 0.0, -0, +0, etc) as zero value.